Add Poky/Yocto support

Circuitsoft opened the pull request:
### What does this PR do?
Adds Poky as a distribution, and supports opkg

### What issues does this PR fix or reference?
Haven't made an issue for this, but it's pretty simple.

### Previous Behavior
Salt does not use opkg on Poky/Yocto images

### New Behavior
Salt uses opkg on Poky/Yocto images

### Tests written?
No

### Commits signed with GPG?
No

Circuitsoft committed:
Add Poky to OS Information Grains

Circuitsoft committed:
Enable opkg on non-NILinuxRT systems

Circuitsoft committed:
Enable opkg as pkgrepo handler on Poky

isbm commented on salt/modules/opkg.py:
```diff
@@ -56,9 +56,11 @@ def __virtual__():
     '''
-    if __grains__.get('os_family', False) == 'NILinuxRT':
+    try:
+        os.listdir(OPKG_CONFDIR)
```
Why would you `os.listdir` it only in order to know that the directory is there? This generates lots of stats for literally nothing. Why not just:

```python
def __virtual__():
    return not os.path.isdir(OPKG_CONFDIR) and (False, "Not OpenEmbedded system") or True

Circuitsoft replied:
Ah. Forgot about that method. Will do.

isbm commented on salt/states/pkgrepo.py:
```diff
@@ -344,7 +344,7 @@ def managed(name, ppa=None, **kwargs):
             else salt.utils.is_true(enabled)
 
-    elif os_family == 'nilinuxrt':
+    elif os_family in ('nilinuxrt', 'poky'):
```
It is "NILinuxRT" or "lilinuxrt"?

Circuitsoft replied:
Previously in that code (not by me) the os_family variable is "normalized" with .lower().

isbm replied:
Could you please fix this? Such "normalisation" usually leads to a nasty bugs than later we have all over the place.

Circuitsoft replied:
So, it seems that the .lower() is originally from e7fb3095ceeecc1e5507754515a34e2d498a6e75. Was there a reason for it then?

cachedout reviewed (changes requested):
I'd like to see the changes recommended by @isbm go in before we merge this.

Circuitsoft committed:
Un-normalize os_family in pkgrepo state

The __grains__['os'].lower() code was added in
e7fb3095ceeecc1e5507754515a34e2d498a6e75, but I don't know why. I hope
this "fix" doesn't break anything.

cachedout commented:
@isbm Re-review requested.

isbm commented on salt/modules/opkg.py:
```diff
@@ -56,9 +56,9 @@ def __virtual__():
+    if os.path.isdir(OPKG_CONFDIR):
         return __virtualname__
-    return (False, "Module opkg only works on nilrt based systems")
+    return (False, "Module opkg only works on OpenEmbedded based systems")
```
These parenthesis aren't needed.

isbm commented on salt/states/pkgrepo.py:
```diff
@@ -333,7 +332,7 @@ def managed(name, ppa=None, **kwargs):
             else salt.utils.is_true(disabled)
 
-    elif os_family in ('redhat', 'suse'):
+    elif __grains__['os_family'] in ('RedHat', 'Suse', 'SUSE'):
```
The all uppercase `os_family` "SUSE" is invalid, it is only 'Suse'. If someone is using something else, then this is wrong.

Circuitsoft replied:
I was looking at the grains source to figure out what all possible values could be, because I didn't know the exact capitalization since the original code used .lower().

isbm commented on salt/states/pkgrepo.py:
```diff
@@ -412,14 +411,14 @@ def managed(name, ppa=None, **kwargs):
                     break
             else:
-                if os_family in ('redhat', 'suse') \
+                if __grains__['os_family'] in ('RedHat', 'Suse', 'SUSE') \
```
Same here. It is "Suse".

Circuitsoft replied:
Ok. I'll change that, and hope I don't break anything.

cachedout commented:
@isbm Can you please take another look here?

isbm reviewed (approved):
@Circuitsoft thanks, well done! @cachedout sorry, I was on a short vacations. :wink:

Commit:
Merge branch '2017.7' into 2017.7
